dali/c_api/c_api.h

```diff
@@ -23,17 +23,23 @@
 #ifdef __cplusplus
 extern "C" {
 #endif
-  struct daliPipelineHandle {
+  /**
+   * @brief Handle for DALI C-like API.
+   *
+   * @note Beware, the C API is just C-like API for handling some mangling issues and
+   * it can throw exceptions.
+   */
+  typedef struct daliPipelineHandle {
```

Comment:
```suggestion
  typedef struct {
```
I don't think anyone's been using it as `struct daliPipelineHandle` - and this repetition is just asking for some compiler to start to complain.

Reply:
There's a slight difference with typedefing an anonymous and named struct (and some people discourage the practice totally), but I don't think we rely on those properties anywhere.

Reply:
Done